sky/packages/sky/lib/src/fn3/framework.dart

```diff
@@ -712,11 +719,11 @@ abstract class BuildableElement<T extends Widget> extends Element<T> {
   /// components dirty during event handlers before the frame begins, not during
   /// the build itself.
   void markNeedsBuild() {
-    assert(!_debugStateLocked);
     assert(_debugLifecycleState == _ElementLifecycle.mounted);
     if (_dirty)
       return;
     _dirty = true;
+    assert(!_debugInBuildScope);
```

Comment:
Seems reasonable. Somewhere, maybe in build's documentation, maybe in setState()'s, maybe StatefulComponent's, not sure, we should say something like "For convenience, calls to setState() during a component's own build are ignored."

Do we catch setState() being invoked on an already-dirty child? If not, that would fail when the child is not dirty but not fail when it's already dirty, which would lead to some really hard-to-nail-down bugs.